chirp/drivers/tdh8.py

```diff
@@ -0,0 +1,2193 @@
+                        list_val = []
+                        lenth_val = 0
+                        while lenth_val < (len(val)):
+                            if val[lenth_val] != ' ':
+                                list_val.append(int(val[lenth_val], 16))
+                                lenth_val += 1
+                            else:
+                                list_val.append(0xFF)
+                                lenth_val += 1
+                        self._memobj.group6.group6 = list_val
+
+                    elif setting == 'group7' and element.value.get_mutable():
+                        val = str(element.value)
+                        list_val = []
+                        lenth_val = 0
+                        while lenth_val < (len(val)):
+                            if val[lenth_val] != ' ':
+                                list_val.append(int(val[lenth_val], 16))
+                                lenth_val += 1
+                            else:
+                                list_val.append(0xFF)
+                                lenth_val += 1
+                        self._memobj.group7.group7 = list_val
+
+                    elif setting == 'group8' and element.value.get_mutable():
+                        val = str(element.value)
+                        list_val = []
+                        lenth_val = 0
+                        while lenth_val < (len(val)):
+                            if val[lenth_val] != ' ':
+                                list_val.append(int(val[lenth_val], 16))
+                                lenth_val += 1
+                            else:
+                                list_val.append(0xFF)
+                                lenth_val += 1
+                        self._memobj.group8.group8 = list_val
+                    elif element.value.get_mutable():
+                        setattr(obj, setting, element.value)
+                except Exception as e:
+                    LOG.debug(element.get_name())
+                    raise
+
+    def _set_fm_preset(self, settings):
+        for element in settings:
+            try:
+                val = element.value
+                if self._memobj.fm_presets <= 108.0 * 10 - 650:
+                    value = int(val.get_value() * 10 - 650)
+                else:
+                    value = int(val.get_value() * 10)
+                LOG.debug("Setting fm_presets = %s" % (value))
+                self._memobj.fm_presets = value
+            except Exception as e:
+                LOG.debug(element.get_name())
+                raise
+
+
+@directory.register
+class TDH8_HAM(TDH8):
+    VENDOR = "TIDRADIO"
+    MODEL = "TD-H8-HAM"
+    ident_mode = b'P31185\xff\xff'
+    _ham = True
+
+
+@directory.register
+class TDH8_GMRS(TDH8):
+    VENDOR = "TIDRADIO"
+    MODEL = "TD-H8-GMRS"
+    ident_mode = b'P31184\xff\xff'
+    _gmrs = True
+    _vhf_range = (136000000, 175000000)
+    _uhf_range = (400000000, 521000000)
```

Comment:
Put your `validate_memory()` routine here and then it will only run for this model.